docs/pages/[...id].js

```diff
@@ -54,17 +58,18 @@ export async function getStaticProps(context: {| params: {| id: string |} |}): P
     props: {
       meta,
       content: mdxSource,
-      pageSourceUrl: `https://github.com/pinterest/gestalt/tree/master/docs/pages/markdown/${id}.md`,
+      pageSourceUrl: `https://github.com/pinterest/gestalt/tree/master/docs/pages/markdown/${pathName}.md`,
     },
   };
 }
 
 export async function getStaticPaths(): Promise<{|
-  paths: {| params: {| id: string |} |}[],
+  paths: {| params: {| id: string | string[] |} |}[],
```

Comment:
Before you land this, pull in the changes from #2302 (once it's merged) and re-run `yarn` to lint for the array Flow type syntax here